x-pack/plugins/spaces/public/share_saved_objects_to_space/components/selectable_spaces_control.tsx

```diff
@@ -22,67 +22,91 @@ import {
 import { i18n } from '@kbn/i18n';
 import { FormattedMessage } from '@kbn/i18n/react';
 import { NoSpacesAvailable } from './no_spaces_available';
-import { useKibana } from '../../../../../../src/plugins/kibana_react/public';
 import { ALL_SPACES_ID, UNKNOWN_SPACE } from '../../../common/constants';
 import { DocumentationLinksService } from '../../lib';
 import { SpaceAvatar } from '../../space_avatar';
-import { SpaceTarget } from '../types';
+import { ShareToSpaceTarget } from '../../types';
+import { useSpaces } from '../../spaces_context';
+import { ShareOptions } from '../types';
 
 interface Props {
-  spaces: SpaceTarget[];
-  selectedSpaceIds: string[];
+  spaces: ShareToSpaceTarget[];
+  shareOptions: ShareOptions;
   onChange: (selectedSpaceIds: string[]) => void;
+  enableCreateNewSpaceLink: boolean;
+  enableSpaceAgnosticBehavior: boolean;
 }
 
 type SpaceOption = EuiSelectableOption & { ['data-space-id']: string };
 
 const ROW_HEIGHT = 40;
-const partiallyAuthorizedTooltip = {
-  checked: i18n.translate(
-    'xpack.spaces.management.shareToSpace.partiallyAuthorizedSpaceTooltip.checked',
-    { defaultMessage: 'You need additional privileges to deselect this space.' }
-  ),
-  unchecked: i18n.translate(
-    'xpack.spaces.management.shareToSpace.partiallyAuthorizedSpaceTooltip.unchecked',
-    { defaultMessage: 'You need additional privileges to select this space.' }
-  ),
-};
-const partiallyAuthorizedSpaceProps = (checked: boolean) => ({
-  append: (
-    <EuiIconTip
-      content={checked ? partiallyAuthorizedTooltip.checked : partiallyAuthorizedTooltip.unchecked}
-      position="left"
-      type="iInCircle"
-    />
-  ),
-  disabled: true,
-});
-const activeSpaceProps = {
-  append: <EuiBadge color="hollow">Current</EuiBadge>,
-  disabled: true,
-  checked: 'on' as 'on',
-};
+const APPEND_ACTIVE_SPACE = (
+  <EuiBadge color="hollow">
+    {i18n.translate('xpack.spaces.shareToSpace.currentSpaceBadge', { defaultMessage: 'Current' })}
```

Comment:
__optional nit:__ we used to prefer `FormattedMessage` whenever possible inside of React tree/components. It's a bit more feature-rich, but in this particular case it doesn't really matter - up to you.

Reply:
Yeah, this was intentional, using i18n.translate() for some of these made the unit tests for the flyout a lot cleaner. Good to know though.